packages/waku/tests/create-pages.test.ts

```diff
@@ -1581,3 +1581,126 @@ describe('createPages - exactPath', () => {
+      createPage({
+        render: 'static',
+        path: '/(group)/test',
+        component: TestPage,
+      }),
+    ]);
+    const { getRouteConfig, handleRoute } = injectedFunctions();
+    expect(await getRouteConfig()).toEqual([
+      {
+        elements: {
+          'page:/test': { isStatic: true },
+        },
+        rootElement: { isStatic: true },
+        routeElement: { isStatic: true },
+        noSsr: false,
+        path: [{ name: 'test', type: 'literal' }],
+      },
+    ]);
+    const route = await handleRoute('/test', {
+      query: '?skip=[]',
+    });
+    expect(route).toBeDefined();
+    expect(route.rootElement).toBeDefined();
+    expect(route.routeElement).toBeDefined();
+    expect(Object.keys(route.elements)).toEqual(['page:/test']);
+  });
+
+  it('path with nested group', async () => {
+    const TestPage = () => null;
+    createPages(async ({ createPage }) => [
+      createPage({
+        render: 'static',
+        path: '/(a)/test/(b)/foo',
+        component: TestPage,
+      }),
+    ]);
+    const { getRouteConfig, handleRoute } = injectedFunctions();
+    expect(await getRouteConfig()).toEqual([
+      {
+        elements: {
+          'page:/test/foo': { isStatic: true },
+        },
+        rootElement: { isStatic: true },
+        routeElement: { isStatic: true },
+        noSsr: false,
+        path: [
+          { name: 'test', type: 'literal' },
+          { name: 'foo', type: 'literal' },
+        ],
+      },
+    ]);
+    const route = await handleRoute('/test/foo', {
+      query: '?skip=[]',
+    });
+    expect(route).toBeDefined();
+    expect(route.rootElement).toBeDefined();
+    expect(route.routeElement).toBeDefined();
+    expect(Object.keys(route.elements)).toEqual(['page:/test/foo']);
+  });
+
+  it('fails when group path collides with literal', async () => {
+    const TestPage = () => null;
+    createPages(async ({ createPage }) => [
+      createPage({
+        render: 'static',
+        path: '/(group)/test',
+        component: TestPage,
+      }),
+      createPage({
+        render: 'static',
+        path: '/test',
+        component: TestPage,
+      }),
+    ]);
+    const { getRouteConfig } = injectedFunctions();
+    await expect(getRouteConfig).rejects.toThrowError('Duplicated path: /test');
+  });
+
+  it('supports grouped path with layout', async () => {
+    const TestPage = () => null;
+    const TestLayout = ({ children }: PropsWithChildren) => children;
+    createPages(async ({ createPage, createLayout }) => [
+      createLayout({
+        render: 'static',
+        path: '/(group)',
+        component: TestLayout,
+      }),
+      createPage({
+        render: 'static',
+        path: '/(group)/test',
+        component: TestPage,
+      }),
+    ]);
+    const { getRouteConfig, handleRoute } = injectedFunctions();
+    expect(await getRouteConfig()).toEqual([
+      {
+        elements: {
+          'layout:/(group)': { isStatic: true },
```

Comment:
@dai-shi is this correct or does it need to be `layout:/`?

Reply:
Yes, because client caches elements based on ids.

You want to add a test case for:
- createLayout `path: '/'` and
- createLayout `path: '/(group)'`

Reply:
Also, add a case for
- createPage `path: '/(group)'` which should fail?

Reply:
> createPage path: '/(group)' which should fail?

This would likely be a duplicate of `/` since that page is usually present. It shouldn't fail on its own though because in theory you could have 
```
/(home)/index
/(home)/other
/foo
```
since there is no /index, there is no duplicate error here.